network/network.go

```diff
@@ -1230,12 +1272,20 @@ func (n *network) disconnected(p *peer) {
 	if !ip.IsZero() {
 		str := ip.String()
 
+		// Remove [str] from [disconnectedIPs] and [connectedIPs]
+		// before calling track on [ip].
 		delete(n.disconnectedIPs, str)
 		delete(n.connectedIPs, str)
 
-		n.track(ip)
+		// If this peer was a validator, call track to attempt to
+		// reconnect with an exponential backoff.
+		if n.vdrs.Contains(p.id) {
+			n.track(ip)
```

Comment:
Do we only want to reconnect to validators and not all peers?

Reply:
Yes, the public IPs in the peer list messages should only be validators. If the peer is not a validator, they should be creating the connection.